src/logic/calculate.js

```diff
@@ -22,29 +23,49 @@ const calculate = (data, btnName) => {
   }
 
   switch (btnName) {
-    case '+':
-    case '-':
-    case 'X':
-    case '/':
-    case '=':
-      if (total && next) {
-        total = operate(total, next, operation);
-      } else {
-        total = next;
-        next = null;
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+    case '0':
+      if (!total && !operation) {
+        const newNext = next ? `${next}${btnName}` : btnName;
+        return { total, next: newNext, operation };
       }
-      break;
-    case '%':
-      if (next) {
-        next = operate(next, 0, btnName);
-      } else if (total) {
-        total = operate(total, 0, btnName);
+      if (total && operation) {
+        const newNext = next ? `${next}${btnName}` : btnName;
+        return { total, next: newNext, operation };
+      }
+      if (operation) {
+        const newTotal = next;
+        const newNext = next ? `${next}${btnName}` : btnName;
+        return { total: newTotal, next: newNext, operation: btnName };
       }
       break;
+    case '+':
+    case '-':
+    case 'X':
+    case '÷': {
+      const newOp = btnName;
+      return { total, next, operation: newOp };
+    }
```

Comment:
- [ ] [REQUIRED] I cannot perform any operation with these operators. Please fix it.
![1](https://user-images.githubusercontent.com/58552748/128587036-ed268ccd-1462-419f-893b-a374e30da77b.gif)